examples/filter_int_iterator.cpp

```diff
@@ -0,0 +1,70 @@
+// examples/filter_int_iterator.cpp -*-C++-*-
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// [P2727](https://wg21.link/P2727) example:
+// An iterator that allows filtering int elements of a sequence.
+#include <beman/iterator_interface/iterator_interface.hpp>
+
+#include <algorithm>
+#include <array>
+#include <iostream>
+
+// filtered_int_iterator uses std::iterator_interface to define a forward iterator
+// that iterates over a sequence of integers, skipping those that do not satisfy a predicate.
+template <typename Pred>
+struct filtered_int_iterator
+    : beman::iterator_interface::ext_iterator_interface_compat<filtered_int_iterator<Pred>,
+                                                               std::forward_iterator_tag,
+                                                               int> {
+    // Default constructor creates an end-of-range iterator.
+    filtered_int_iterator() : m_it_begin(nullptr) {}
+
+    // Constructor for the beginning of the sequence.
+    filtered_int_iterator(int* it_begin, int* it_end, Pred pred) : m_it_begin(it_begin), m_it_end(it_end), m_pred(std::move(pred)) {
+        m_it_begin = std::find_if(m_it_begin, m_it_end, m_pred);
+    }
+
+    // A forward iterator based on iterator_interface usually requires
+    // three user-defined operations.  since we are adapting an existing
```

Comment:
```suggestion
    // three user-defined operations.  since we are adapting an existing
```

maybe mentions which operations?